core-requirements.txt

```diff
@@ -2,7 +2,7 @@ numpy>=1.20.0
 numba==0.53
 pandas>=1.3.0
 scipy>=1.5.0
-scikit-learn>=0.24.0,<1.0
+scikit-learn>=0.24.0
```

Comment:
Note:
scikit-learn >1.0 will require scikit-optimize>=0.9.0 because modules / objects get shuffled around, causing the following:
```
evalml/pipelines/components/estimators/classifiers/logistic_regression_classifier.py:4: in <module>
    from skopt.space import Real
../evalml_venv/lib/python3.7/site-packages/skopt/__init__.py:55: in <module>
    from .searchcv import BayesSearchCV
../evalml_venv/lib/python3.7/site-packages/skopt/searchcv.py:16: in <module>
    from sklearn.utils.fixes import MaskedArray
E   ImportError: cannot import name 'MaskedArray' from 'sklearn.utils.fixes' (/Users/angela.lin/Desktop/evalml_venv/lib/python3.7/site-packages/sklearn/utils/fixes.py)```

Our tests, which automatically install the latest version of scikit-optimize (0.9.0) will thus not break, so maybe we don't need to set our requirement of `scikit-optimize>=0.9.0` but open to discussion if it feels like we should.

Reply:
How did you run into this?

Reply:
My local environment uses `scikit-optimize==0.8.1` so as soon as I tried using scikit-learn 1.0, things broke 🥲

Reply:
I see. I'm wondering if this will affect users in some way. Did you just install scikit-learn or did you do `pip install -r requirements.txt` ?

Reply:
I just installed `scikit-learn`! Our tests didn't break because they call `pip install -r requirements.txt` and in a clean environment, 0.9.0 will be installed.

This is a known issue in the scikit-optimize repo where it seems like the community solution is to either use scikit-learn<=0.24.2 or use scikit-optimize>=0.9.0: https://github.com/scikit-optimize/scikit-optimize/issues/569

Reply:
Cool, makes sense to me!

Reply:
Would it be simpler to make the core requirements use `scikit-learn >= 1.0.0` versus having support for both but test coverage for the later version?

Reply:
Hmm, I can be persuaded otherwise but I think this aligns with what we've currently been doing aka every time our dependencies come out with a new version, we pin until we can fix the tests to support the latest version--and thats the version that we run on!